<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">

	<div class="item"  id="item1">
		<h1>How to run quan?</h1>
		<p>
		The QTLtools mode quant allows quantifying exons and genes and requires 2 files as input:  
		</p>

		<ul>
			<li>A <i>BAM</i> file containing the sequence data for a RNA-seq experiment</li>
			<li>A <i>GTF</i> file containing the exon and gene annotations that can be downloaded from <a href="http://www.gencodegenes.org/releases/current.html">here</a></li>
		</ul>
		
		<p>
		To illustrate how this works, we provide the 4 following example files:
		</p>

		<ul>
			<li>A RNA-seq <i>BAM</i> file on chr22 for sample HG00381: <a href="http://jungle.unige.ch/QTLtools_examples/HG00381.chr22.bam" download>BAM</a> / <a href="http://jungle.unige.ch/QTLtools_examples/HG00381.chr22.bam" download>index</a></li>
			<li>A <i>GTF</i> file containing the gencode annotation v19 for chr22: <a href="http://jungle.unige.ch/QTLtools_examples/gencode.v19.annotation.chr22.gtf.gz" download>GTF</a></li>
		</ul>

		<p>
		Then, to perform quantification from paired-end RNA-seq, we recommand to use this command:
		</p>

		<code>
		QTLtools <b>quan</b> --bam HG00381.chr22.bam --gtf gencode.v19.annotation.chr22.gtf.gz --samples HG00381 --out-prefix HG00381 --filter-mapping-quality 150 --filter-mismatch 5 --filter-mismatch-total 5 --rpkm --no-merge
		</code>


		<p>
		This command produces 4 output files <i>HG00381.exon.count.bed</i>, <i>HG00381.exon.rpkm.bed</i>, <i>HG00381.gene.count.bed</i> and <i>HG00381.gene.rpkm.bed</i>. If you open the latter, you'll get something like this: 
		</p>

		<code>
		#chr	start	end	gene	length	strand	HG00381<br>
		22	16062156	16062157	ENSG00000233866.1	586	+	0<br>
		22	16076171	16076172	ENSG00000229286.1	121	-	0<br>
		22	16084825	16084826	ENSG00000235265.1	578	-	0<br>
		22	16122719	16122720	ENSG00000215270.3	1049	+	56.8023992<br>
		22	16124972	16124973	ENSG00000223875.1	685	-	0<br>
		22	16151396	16151397	ENSG00000271672.1	622	-	8.708815662<br>
		...<br>
		22	51181947	51181948	ENSG00000254499.1	284	-	0<br>
		22	51193861	51193862	ENSG00000213683.3	767	-	953.4279676<br>
		22	51222090	51222091	ENSG00000079974.13	5870	-	90.43519038<br>
		22	51222184	51222185	ENSG00000184319.11	3603	+	79.68215851
		</code>

		<p>
		This file is a <i>BED</i> file with the following 7 columns:
		</p>

		<ul>
			<li>1. The chromosome ID</li>
			<li>2. The TSS position [0-based]</li>
			<li>3. The TSS position [1-based]</li>
			<li>4. The gene ID</li>
			<li>5. The physical length of the gene in bp</li>
			<li>6. The strand orientation</li>
			<li>7. The RPKM value for the sample (here HG00381) at the corresponding gene</li>
		</ul>

		<p>
		Now, if you open <i>HG00381.exon.rpkm.bed</i>, you'll get:
		</p>

		<code>
		#chr	start	end	gene	length	strand	HG00381<br>
		...<br>		
		22      17517459        17517460        ENSG00000237438.2_17517460_17518234     ENSG00000237438.2       +       7.874866966<br>
		22      17517459        17517460        ENSG00000237438.2_17525763_17525890     ENSG00000237438.2       +       106.9270201<br>
		22      17517459        17517460        ENSG00000237438.2_17528214_17528320     ENSG00000237438.2       +       77.96262006<br>
		...<br>
		22      17565843        17565844        ENSG00000177663.9_17565844_17566119     ENSG00000177663.9       +       3.009379634<br>
		22      17565843        17565844        ENSG00000177663.9_17578687_17578833     ENSG00000177663.9       +       166.3142867<br>
		22      17565843        17565844        ENSG00000177663.9_17579665_17579777     ENSG00000177663.9       +       21.7314494<br>
		...
		</code>

		<p>
		For exon quantifications, the output is slightly different. The 4th and the 5th columns give this time, the exonID and the gene it belongs to, respectively. The TSS position is the same for all exons within the same gene. These pieces of information are important for QTL mapping in cis.
		</p>

		<p> 
		To quantify all samples for which you have RNA-seq, proceed as follows:
		</p>

		<ul>
			<li>A. Run <i>QTLtools quan</i> for each BAM file separately (i.e. in indepeneden jobs)</li>
			<li>B. Extract the annotation columns using <i>cut -f1-6</i> on one of the file</li>
			<li>C. Extract the per sample quantifications using <i>cut -f7</i> for each file separately </li>
			<li>D. Build the entire quantification matrix using the <i>paste</i> to merge B with all quantifications of C</li>
			<li>E. Filter all poorly quantified exons/genes using <i>awk</i> for example</li>
		</ul>

		<p>
		The expected results will look like this:
		</p>

		<code>
		#chr    start   end     gene    length  strand  HG00096 HG00097 HG00099 HG00100 HG00101 HG00102<br>
		chr1       29369   29370   ENSG00000227232.4       1800    -       3.574873147     2.159108572     2.656253435     2.147152811     3.060300881     0.4603702761<br>
		chr1       135894  135895  ENSG00000268903.1       755     -       1.094681369     0.5444517286    0       0       1.015910092     0<br>
		chr1       137964  137965  ENSG00000269981.1       284     -       2.494419821     1.184234794     0.7113553138    0.9226247455    3.437314236     2.122065293<br>
		chr1       139378  139379  ENSG00000237683.5       2661    -       1.575143469     1.306025695     0.2530689078    0.3692579309    3.092051181     1.358887358<br>
		chr1       267252  267253  ENSG00000228463.4       3954    -       0.07465174813   0.09450983011   0       0.1822382211    0.03526978984   0.209576757<br>
		chr1       317719  317720  ENSG00000237094.7       6869    +       0       0.005440265952  0       0.009536520153  0       0.03290143452<br>
		chr1       564441  564442  ENSG00000225972.1       372     +       1269.561342     1101.687828     1226.812553     1438.322375     1157.266087     2486.809527<br>
		chr1       565019  565020  ENSG00000225630.1       1044    +       1324.772668     1573.121065     1388.501978     1507.901121     1498.158208     2563.208926<br>
		chr1       566453  566454  ENSG00000237973.1       1543    +       5898.256933     6113.821775     5275.691308     6484.280259     5137.033507     8913.352248
		</code>

		<p>
		Hereafter the list of options that can be used to tune the quantification step:
		</p>

		<ul>
			<li><i>--out-prefix</i>: Output file prefix</li>
    			<li><i>--rpkm</i>: Print RPKM values</li>
			<li><i>--gene-types</i>: Gene types to quantify. (Requires gene_type attribute in GTF. It will also use transcript_type if present) </li>
			<li><i>--max-read-length [1000]</i>: Group genes separated by this much together. Set this larger than your read length</li>
			<li><i>--filter-mapping-quality [10]</i>: Minimal phred mapping quality for a read to be considered</li>
			<li><i>--filter-mismatch [OFF]</i>: Maximum mismatches allowed in a read. If between 0 and 1 taken as the fraction of read length (Requires NM attribute)</li>
			<li><i>--filter-mismatch-total [OFF]</i>: Maximum total mismatches allowed in paired reads. If between 0 and 1 taken as the fraction of combined read length (Requires NM attribute)</li>
			<li><i>--check-proper-pairing</i>: If provided only properly paired reads according to the aligner that are in correct orientation will be considered. Otherwise all pairs in correct orientation will be considered</li>
			<li><i>--check-consistency</i>: If provided checks the consistency of split reads with annotation, rather than pure overlap of one of the blocks of the split read</li>
        		<li><i>--no-merge</i>: If provided overlapping mate pairs will not be merged</li>
			<li><i>--filter-remove-duplicates</i>: Remove duplicate sequencing reads in the process</li>
		</ul>

	</div>
	
	<div class="log">Monday 11th July 2016</div>
</div>
</body>
</html>
