<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=iso-8859-1"/>
<meta http-equiv="Cache-Control" content="max-age=3600"/>
<meta name="description" content="description"/>
<meta name="keywords" content="keywords"/> 
<meta name="author" content="author"/> 
<link rel="stylesheet" type="text/css" href="../style/default.css" media="screen"/>
<title>QTLtools</title>
<script language="Javascript" src="../script/print_last_modif_date.js" ></script>
</head>

<body>
<div class="content">

	<div class="item"  id="item1">
		<h1>How to run a full pass in trans?</h1>
		<p>
		In a full analysis in <i>trans</i>, you need to run separately the nominal pass and the permutation passes.
		To illustrate this, first download the example data below and go through the two steps; 1. nominal pass and 2. permutation pass.
		</p>

		<ul>
			<li> The phenotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genes.50percent.chr22.bed.gz.tbi" download>index</a></li>
			<li> The genotype data matrix for chr22 on 358 samples: <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz" download>BED</a> / <a href="http://jungle.unige.ch/QTLtools_examples/genotypes.chr22.vcf.gz.csi" download>index</a></li>
		</ul>

		<br>
		<h3>Step1: Run the nominal pass</h3>
		<p>
		To do so, run:
		</p>

		<code>
		QTLtools <b>trans</b> --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --threshold 1e-5 <b>--nominal</b> --out trans.nominal
		</code>

		<p>
		This computes all pairwise test between variants and phenotypes (excluding the cis-window) and reports the results in 3 output files:
		</p>

		<ul>
			<li><i>trans.nominal.best.txt.gz</i>; contains the top hit per phenotype</li>
			<li><i>trans.nominal.bins.txt.gz</i>; contains a binning of all hits with a P-value above the specified <i>--threshold</i> (=1e-5)</li>
			<li><i>trans.nominal.hits.txt.gz</i>; contains detailed information (not binned) for all hits with a P-value <b>below</b> the specified <i>--threshold</i> (=1e-5)</li>
		</ul>

		<p>
		These files are detailed below:
		</p>

		<code>
		zcat trans.nominal.best.txt.gz | head<br>
		ENSG00000215270.3 -1 1.12926e-05<br>
		ENSG00000237438.2 -1 2.03908e-05<br>
		ENSG00000273203.1 -1 0.000138072<br>
		ENSG00000273442.1 -1 1.64291e-05<br>
		ENSG00000177663.9 -1 3.32289e-05<br>
		ENSG00000183307.3 -1 5.1272e-06<br>
		ENSG00000069998.8 -1 3.91133e-06<br>
		ENSG00000093072.11 -1 1.82101e-06<br>
		ENSG00000099954.14 -1 4.99988e-07<br>
		ENSG00000182902.9 -1 9.67087e-06<br>
		</code>
		
		<ul>
			<li>1. Phenotype ID </li>
			<li>2. Dummy</li>
			<li>3. Nominal P-value of the top hit per phenotype</li>
		</ul>		

		<code>
		zcat trans.nominal.bins.txt.gz | head<br>
		0 0 0.000231095 1 0.996523 147445<br>
		1 0.000231095 0.000462191 0.996523 0.993047 148408<br>
		2 0.000462191 0.000693286 0.993047 0.989571 149815<br>
		3 0.000693286 0.000924382 0.989571 0.986094 149084<br>
		4 0.000924382 0.00115548 0.986094 0.982619 147263<br>
		5 0.00115548 0.00138657 0.982619 0.979143 147749<br>
		6 0.00138657 0.00161767 0.979143 0.975668 149862<br>
		7 0.00161767 0.00184876 0.975668 0.972193 149279<br>
		8 0.00184876 0.00207986 0.972193 0.968719 149383<br>
		9 0.00207986 0.00231095 0.968719 0.965245 147444<br>
		</code>

		<ul>
			<li>1. Index of the bin </li>
			<li>2-3. Dummy</li>
			<li>4-5. Boundaries of the P-value bin</li>
			<li>6. Number of tests falling in this bin</li>
		</ul>

		<code>
		zcat trans.nominal.hits.txt.gz | sort -k7,7g | head<br>
		ENSG00000207833.1 chr22 23134977 rs147092127 chr22 39863524 9.48389e-13 -1 0.365398<br>
		ENSG00000207833.1 chr22 23134977 rs76864396 chr22 39888462 1.19178e-12 -1 0.363899<br>
		ENSG00000207833.1 chr22 23134977 rs12628516 chr22 39939913 2.00021e-12 -1 0.360471<br>
		ENSG00000211649.2 chr22 22723982 rs141368136 chr22 32710808 1.16936e-10 -1 0.331991<br>
		ENSG00000100033.12 chr22 18924066 rs5770344 chr22 49779694 1.45997e-10 -1 0.330351<br>
		ENSG00000211668.2 chr22 23134980 rs147092127 chr22 39863524 1.69775e-10 -1 0.32923<br>
		ENSG00000211668.2 chr22 23134980 rs76864396 chr22 39888462 1.90403e-10 -1 0.328375<br>
		ENSG00000211668.2 chr22 23134980 rs12628516 chr22 39939913 2.57114e-10 -1 0.326123<br>
		ENSG00000211660.3 chr22 23040274 rs73172520 chr22 47653458 5.40445e-10 -1 0.320473<br>
		ENSG00000100033.12 chr22 18924066 rs5770381 chr22 49788937 6.03097e-10 -1 0.319629<br>
		</code>
	
		<ul>
			<li>1. Phenotype ID</li>
			<li>2. Phenotype chrID</li>
			<li>3. Phenotype start</li>
			<li>4. Variant ID</li>
			<li>5. Variant chrID</li>
			<li>6. Variant position</li>
			<li>7. Nominal P-value of association</li>
			<li>8. Dummy</li>
			<li>9. Regression slope</li>
		</ul>

		<p>
		This trans nominal pass can be tuned using the following options:
		</p>
		<ul>
			<li><i>--threshold [float]</i>: this defines the threshold to bin (in *.bins.txt.gz files) or report (in *.hits.txt.gz) nominal P-values. Default is 1e-5.</li>
			<li><i>--normal</i>: this enforce the phenotypes to be normally distributed</li>
			<li><i>--window</i>: this defines the <i>cis</i>-window of variants to be excluded from the analysis</li>
			<li><i>--bins [int]</i>: this defines the number of bins to be used for binning P-values above --threshold (default is 1000)</li>
		</ul>

		<br>
		<h3>Step2: Run the permutation passes</h3>

		<p>
		The permutation pass is relatively similar to the nominal one. For a single permutation, run:
		</p>
		
		<code>
		QTLtools <b>trans</b> --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --threshold 1e-5 <b>--permute</b> --out trans.perm123 --seed 123
		</code>

		<p>
		This command permutes all the phenotypes in a similar way so that correlation between phenotypes is unchanged.
		All pairwise tests between variants and phenotypes are then computed (excluding the cis-window) and reported in 3 files as done by the nominal pass: <i>trans.perm123.best.txt.gz</i>, <i>trans.perm123.bins.txt.gz</i> and <i>trans.perm123.hits.txt.gz</i>.
		</p>

		<note>
		YOU NEED TO CHANGE THE SEED FOR EACH PERMUTATION! Please, change the value of <i>--seed</i> each time you run the command.
		</note>

		<p>
		For instance, to run 20 permutations using 4 threads, you can proceed as follows:
		</p>

		<code>
		for p in $(seq 1 20); do<br>
		&nbsp;&nbsp;&nbsp;&nbsp;echo "trans --vcf genotypes.chr22.vcf.gz --bed genes.50percent.chr22.bed.gz --permute --out trans.perm$p --seed $p";<br>
		done | xargs -P4 -n10 QTLtools
		</code>

		<br>
		<h3>Produce a QQplot and estimate FDR</h3>

		<p>
		Once the nominal and permutation passes done, you can produce a QQplot by running:
		</p>
		
		<code>
		Rscript ~/script/plotTrans.R QQplot.pdf trans.nominal.hits.txt.gz trans.perm*.hits.txt.gz
		</code>

		<p>
		For the example data, you should get something like this:
		</p>

		<img WIDTH=25% src="../img/fastqtl_qqplot.png"></img>

		<p>
		Now, to estimate FDR, you can go through all the best nominal hits and count how many hits in the permutation passes give a smaller P-values.
		For instance, for the 100th smallest nominal P-values, if you get 10 smaller permutation P-values, you can assume a 10% FDR at that level.
		We are currently working on a R script to do this automatically and will release it ASAP. 
		</p>
	</div>
	
	<div class="log">Tuesday 12th July 2016</div>
</div>
</body>
</html>
